src/main/java/seedu/address/model/student/Student.java

```diff
@@ -185,14 +186,20 @@ public Student addAssignment(Assignment assignmentToAdd) {
         newAssignmentlist.add(assignmentToAdd);
         return new Student(this.name, this.phone, this.email, this.tags, newAssignmentlist, this.remark);
     }
+
     /**
-     * Deletes the Assignment at {@code index} in the student's assignmentList
+     * Updates the Assignment at {@code index} to a new Assignment in the student's assignmentList
      */
-    public void deleteAssignment(int index) {
-        if (assignmentList.size() < (index - 1) || index <= 0) {
+    public Student updateAssignment(Index index, Assignment assignmentToUpdate)
+            throws AssignmentIndexOutOfRangeException {
+        int indexZeroBased = index.getZeroBased();
+        if (indexZeroBased < 0 || indexZeroBased >= assignmentList.size()) {
             throw new AssignmentIndexOutOfRangeException();
         }
-        assignmentList.remove(index - 1);
+        List<Assignment> newAssignmentList = new ArrayList<Assignment>(assignmentList);
+        newAssignmentList.set(indexZeroBased, assignmentToUpdate);
+        assignmentList = newAssignmentList;
```

Comment:
Why do we have to create a newAssignmentList to set it? Should we just use the old assignment list to set?

Reply:
Good point. I will fix this.